Sources/EdgeState.swift

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright 2021 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import AEPCore
+import AEPServices
+import Foundation
+
+/// Updates the state of the  `Edge` extension based on the Collect consent status
+class EdgeState {
+    private let LOG_TAG = "EdgeState"
+    private(set) var hitQueue: HitQueuing
+    private(set) var hasBooted = false
+    private(set) var currentCollectConsent: ConsentStatus
+
+    /// Creates a new `EdgeState` and initializes the required properties and sets the initial collect consent
+    init(hitQueue: HitQueuing) {
+        self.hitQueue = hitQueue
+        self.currentCollectConsent = EdgeConstants.Defaults.COLLECT_CONSENT_PENDING
+        hitQueue.handleCollectConsentChange(status: currentCollectConsent)
+    }
+
+    /// Completes init for the `Edge` extension and the collect consent is set based on either Consent shared state
+    /// or the default value if this extension is not registered
+    ///
+    /// - Parameters:
+    ///   - event: The `Event` triggering the bootup
+    ///   - getXDMSharedState: used to fetch the Consent data
+    func bootupIfNeeded(event: Event, getXDMSharedState: @escaping (String, Event?) -> SharedStateResult?) {
+        guard !hasBooted else { return }
+
+        // check if consent extension is registered
+        if let consentSharedState = getXDMSharedState(EdgeConstants.SharedState.Consent.SHARED_OWNER_NAME, nil) {
```

Comment:
I think we should be reading the EventHub shared state to determine if the consent extension is registered. I think this approach of checking if the shared state exists to determine if an extension is registered can lead to false negatives.

We've used this method in our legacy codebase before, but I think we've found it tricky in some situations. Just because shared state doesn't exist for an extension doesn't mean it doesn't exist.

Reply:
I looked at using either the shared state or MobileCore.getRegisteredExtensions API but found that it only includes the friendly name and version of each extension, whereas other extensions should rely on is the extension name (aka shared state name) as the friendly name can anytime change. I discussed a bit about this with @shalehaha and we are considering it, but are still evaluating the implications for Assurance UI before making any changes.

I am willing to change this logic to check for hub shared state/getRegisteredExtensions as soon as the extension name is included in there.